lib/usd/hdMaya/adapters/meshAdapter.cpp

```diff
@@ -0,0 +1,444 @@
+//
+// Copyright 2019 Luma Pictures
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include <pxr/pxr.h>
+
+#include <pxr/base/tf/type.h>
+
+#include <pxr/usd/usdGeom/tokens.h>
+
+#include <pxr/imaging/hd/tokens.h>
+#include <pxr/imaging/pxOsd/tokens.h>
+
+#include <maya/MAnimControl.h>
+#include <maya/MCallbackIdArray.h>
+#include <maya/MDGContext.h>
+#include <maya/MDGContextGuard.h>
+#include <maya/MFloatArray.h>
+#include <maya/MFnMesh.h>
+#include <maya/MIntArray.h>
+#include <maya/MItMeshPolygon.h>
+#include <maya/MNodeMessage.h>
+#include <maya/MObjectHandle.h>
+#include <maya/MPlug.h>
+#include <maya/MPolyMessage.h>
+
+#include "adapterDebugCodes.h"
+#include "adapterRegistry.h"
+#include "mayaAttrs.h"
+#include "shapeAdapter.h"
+#include "tokens.h"
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+namespace {
+
+constexpr int MAX_SMOOTH_LEVEL = 8;
```

Comment:
Why 8? Magic numbers need comments.

Reply:
Agreed. The number is probably the max level defined by Maya OSD implementation?

Reply:
Not totally sure, but I think Pal added this just as a safeguard against accidentally crashing Maya by setting the smooth level too high.  (And I'm guessing he just picked a number that he though would be "high enough.") I decided to get rid of it altogether, though, because you never know, there might be someone who wants it higher than 8... and I don't want to limit end users.

https://github.com/Autodesk/maya-usd/pull/231/commits/b69c0500e211963f11d6a11e2e1feda1528bd1c2